modules/masterbar/masterbar.php

```diff
@@ -68,6 +118,50 @@ function __construct() {
 		add_action( 'wp_logout', array( $this, 'maybe_logout_user_from_wpcom' ) );
 	}
 
+	public function tracks_and_redirect_add_route() {
+		add_rewrite_rule( 'jetpack-track-and-bounce.php\?tracks_and_bounce=(\w*)', 'index.php?tracks_and_bounce=$matches[1]', 'top' );
+	}
+
+	public function tracks_and_redirect_add_query_vars( $query_vars ) {
+		$query_vars[] = 'tracks_and_bounce';
+		$query_vars[] = 'tracks_and_bounce_nonce';
+		return $query_vars;
+	}
+
+	public function tracks_and_redirect_parse_request( &$wp ) {
+		if (
+			array_key_exists( 'tracks_and_bounce', $wp->query_vars ) &&
```

Comment:
Could we use `get_query_var` instead of accessing `$_GET` directly?

Reply:
`get_query_var` is not quite working for this scenario... will look into it later. For the moment I've replaced it with `$wp->query_vars[...]`.

Reply:
Might be because the new values that we are using were not whitelisted before? (`query_vars` filter)